prefect_gcp/cloud_run_job.py

```diff
@@ -0,0 +1,578 @@
+                f"the correct region for your Cloud Run Job and that {self.project_id} is the "
+                "correct GCP project. If your project ID is not correct, you are using a Credentials "
+                "block with permissions for the wrong project."
+            ) from exc
+
+        raise exc
+
+    def _job_run_submission_error(self, exc):
+        """Provides a nicer error for 404s when submitting job runs."""
+        if exc.status_code == 404:
+            pat1 = r"The requested URL [^ ]+ was not found on this server"
+            pat2 = r"Resource '[^ ]+' of kind 'JOB' in region '[\w\-0-9]+' in project '[\w\-0-9]+' does not exist"
+            if re.findall(pat1, str(exc)):
+                raise RuntimeError(
+                    f"Failed to find resources at {exc.uri}. Confirm that region '{self.region}' is "
+                    f"the correct region for your Cloud Run Job and that '{self.project_id}' is the "
+                    "correct GCP project. If your project ID is not correct, you are using a Credentials "
+                    "block with permissions for the wrong project."
+                ) from exc
+            else:
+                raise exc
+
+        raise exc
+
+    @sync_compatible
+    async def run(self, task_status: Optional[TaskStatus] = None):
+        with self._get_jobs_client() as jobs_client:
+            try:
+                self.logger.info(f"Creating Cloud Run Job {self.job_name}")
+                self._create_job(jobs_client=jobs_client, body=self._jobs_body())
+            except googleapiclient.errors.HttpError as exc:
+                self._create_job_error(exc)
+
+            try:
+                self._wait_for_job_creation(client=jobs_client)
+            except Exception as exc:
+                self.logger.exception(
+                    f"Encountered an exception while waiting for job run creation {exc!r}"
+                )
+                if not self.keep_job_after_completion:
+                    try:
+                        self.logger.info(
+                            f"Deleting Cloud Run Job {self.job_name} from Google Cloud Run."
+                        )
+                        self._delete_job(jobs_client=jobs_client)
+                    except Exception as exc:
+                        self.logger.exception(
+                            f"Encountered an exception while attempting to delete failed Cloud Run Job.'{self.job_name}':\n{exc!r}"
+                        )
+                raise exc
+
+            try:
+                self.logger.info(
+                    f"Submitting Cloud Run Job {self.job_name} for execution."
+                )
+                job_execution = Execution.from_json(
+                    self._submit_job_for_execution(jobs_client=jobs_client)
+                )
+
+                command = (
+                    " ".join(self.command)
+                    if self.command
+                    else "'default container command'"
```

Comment:
The command would have double `''default container command''`

```suggestion
                    else "default container command"
```